app/services/BusinessRatesAttachmentsService.scala

```diff
@@ -72,13 +73,48 @@ class BusinessRatesAttachmentsService @Inject()(
 
   def submit(
               submissionId: String,
-              nonEmptyReferences: List[String]
+              nonEmptyReferences: List[String],
+              retryCount: Int = 0
             )(implicit request: LinkingSessionRequest[_], hc: HeaderCarrier): EitherT[Future, AttachmentException, List[Attachment]] = {
     EitherT.fromEither[Future](Either.cond(nonEmptyReferences.nonEmpty, nonEmptyReferences, MissingRequiredNumberOfFiles))
       .semiflatMap(Future.traverse(_)(r => getAttachment(r).map(r -> _)))
+      .subflatMap(attachments => {
+        Either.cond(
+          attachments.filter(
+            attachment =>
```

Comment:
this chunk is repeated twice and so if the states change this will rely on someone remembering to update the list in two different places. , also the contains check is performed twice. you could have a function Attachment => Boolean that has that List of states in it and checks for the contains, then just use the function in these two places